### ISSUE TITLE
GZIP decompression

### ISSUE THREAD

**Issue by wuxianzhong-wuxianzhong** [zeek/zeek#2088]
The accept-encoding in the HTTP request is gzip, but the actual body content is not compressed. Zeek recognizes gzip and uses gzip to process (unzip), resulting in null body. Would you like to be able to handle the gzip processing failure again using plaintext transmission processing? Mainly HTTP. Cc files

**Comment by 0xxon** [comment]
Do you have an example of this happening in practice (like a pcap)?

In general - if I understand you correctly, all the signaling shows that some content should be gzip compressed, but it turns out not to be. These cases always are a bit hard for us, because a well-formed client might also not accept the response that a server is delivering - and fail similarly to us. If this is something that commonly happens I could see that it might be worth putting in a workaround though - or at least raise a notice/weird that something weird is going on.

**Comment by wuxianzhong-wuxianzhong** [comment]
> Do you have an example of this happening in practice (like a pcap)?
> 
> In general - if I understand you correctly, all the signaling shows that some content should be gzip compressed, but it turns out not to be. These cases always are a bit hard for us, because a well-formed client might also not accept the response that a server is delivering - and fail similarly to us. If this is something that commonly happens I could see that it might be worth putting in a workaround though - or at least raise a notice/weird that something weird is going on.

Yes, there is such an example in practice. I can send you a pcap package. You can use zeek to restore it. You can find that the restored request_body is empty, but it is not empty. I'm now hoping you can tell me how I need to change the code if the gzip decompression fails and the decompression needs to 
[pcap.zip](https://github.com/zeek/zeek/files/8631077/pcap.zip)
be attempted again (but not gzip). I read the HTTP.cc file. In the HTTP_Entity::DeliverBody method (line 194), I hope that after judging that the encoding is GZIP, if the processing fails, I will call a processing method whose encoding is not GZIP. do you understand what I mean?

**Comment by rsmmr** [comment]
We are already *not* relying on the content-type header, but do file sniffing instead. Doing that for encoding as well would be consistent with that.

**Comment by wuxianzhong-wuxianzhong** [comment]
> 



> We are already _not_ relying on the content-type header, but do file sniffing instead. Doing that for encoding as well would be consistent with that.

Which version, the latest version? You mean using file sniffing on the processing of the content? Don't care about the value of content-encoding? This problem is not only for the processing of gzip, but it is hoped that after the body content processing fails, the original content will be output instead of returning null or empty.
